src/swupd/manifest.go

```diff
@@ -226,3 +226,113 @@ func (m *Manifest) ReadManifestFromFile(f string) error {
 	// return err so the deferred close can modify it
 	return err
 }
+
+// getManifestTimeStamp returns the unix timestamp of m.Header.TimeStamp
+func (m *Manifest) getManifestTimeStamp() int64 {
+	return m.Header.TimeStamp.Unix()
+}
+
+// writeManifestFileHeader writes the header of a manifest to the file
+func writeManifestFileHeader(m *Manifest, w *bufio.Writer) error {
+	var err error
+	if err = m.CheckHeaderPopulated(); err != nil {
+		return err
+	}
+
+	if _, err = w.WriteString(fmt.Sprintf("MANIFEST\t%v\n", m.Header.Format)); err != nil {
```

Comment:
Worth looking at  [errors are values](https://blog.golang.org/errors-are-values) to see if the idea of errorwriter might help. Of course this sort of negates my comment about io.Writer vs bufio.Writer as bufio.Writer is an errorwriter

Reply:
Interesting, it looks like this means errors in the previous writes will be caught when `w.Flush()` is called later?